src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -256,7 +255,12 @@ protected function resolveExprConstFetch(Expr\ConstFetch $node)
 
     protected function resolveExprClassConstFetch(Expr\ClassConstFetch $node)
     {
-        $refClass     = $this->fetchReflectionClass($node->class);
+        $classToReflect = $node->class;
```

Comment:
I agree we need to test for this case, but it isn't demonstrated in the example above. It would be something like this, which I'm doubting is valid PHP:

```php
const LEFT = 'Foo';
const RIGHT = 'Bar';

class FooBar {
    const FILENAME = 'baz.php';
}

require((LEFT . RIGHT)::FILENAME);
```

Like I said... I'm unsure if this is valid PHP, but I think, given that PhpParser accepts an expression, and we need something that can evaluate at compile time, something like this would be a valid test, even if not legal PHP currently.

Opinions?